### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by gmarintes**
fixed buildPattern() so that EnhancedResourcesItemReader can recursively traverse a directory of files

Adding "/**" (and not File.separator + "**") at the end of a directory makes PathMatchingResourcePatternResolver recursively traverse directories. 

Added a unit test that runs a job that imports *.txt from a directory tree.

### CODE REVIEW

**Comment by sastafford** [comment]
Thanks for the pull request.  

The testing needs to change.  The EnhancedResourceItemReaderTestJobConfig class is not necessary since the only thing I want to test is the EnhancedResourcesItemReader class.   I'll add some comments in the file changes.

File: file/src/test/java/com/marklogic/spring/batch/item/file/EnhancedResourceItemReaderTestJobConfig.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+            file.newFolder("A");
+            file.newFolder("B");
+            file.newFolder("A", "A1");
+
+            file.newFile("/A/red.txt");
+            file.newFile("/B/blue.jpg");
+            file.newFile("/B/yellow.txt");
+            file.newFile("/A/A1/green.txt");
+        }
```
Comment (sastafford) [inline file/src/test/java/com/marklogic/spring/batch/item/file/EnhancedResourceItemReaderTestJobConfig.java]: This try block above should be part of a code snippet before the test is run.  

Given a folder with subfolders
When the ItemReader reads the top level folder
Then the ItemReader recursively navigates the file tree and reads each file.  

So keep this part and add it to your test class.

File: file/src/test/java/com/marklogic/spring/batch/item/file/EnhancedResourcesItemReaderTest.java [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobParametersBuilder;
+import org.springframework.test.context.ContextConfiguration;
 
 import java.io.File;
 
 /**
  * Unit test for verifying we construct the pattern correctly.
  */
-public class EnhancedResourcesItemReaderTest extends Assert {
```
Comment (sastafford) [inline file/src/test/java/com/marklogic/spring/batch/item/file/EnhancedResourcesItemReaderTest.java]: Revert this back to just extend the Assert class.  No need to run an entire job.  Just test the ItemReader.

File: file/src/test/resources/job.properties [hunk 3]
```diff
@@ -0,0 +1,13 @@
+marklogic.host=localhost
```
Comment (sastafford) [inline file/src/test/resources/job.properties]: You will be able to delete this properties file as well.

**Comment by gmarintes** [comment]
Made the changes requested.
EnhancedResourcesItemReaderTest now doesn't run a job to test EnhancedResourcesItemReader.
Removed previous changes made to run the test job.

**Pull request merged**